syntax = "proto3";

option java_multiple_files = false;
option java_package = "com.ewolff.microservice.customer.grpc";
option java_outer_classname = "CustomerProto";

import "google/protobuf/empty.proto";
import "order.proto";

service CustomerService {
  rpc getCustomer(CustomerRequest) returns (CustomerResponse);
  rpc getCustomerList(google.protobuf.Empty) returns (CustomerListResponse);
  rpc CreateCustomer(CreateCustomerRequest) returns (CustomerResponse);
  rpc UpdateCustomer(UpdateCustomerRequest) returns (CustomerResponse);
  rpc DeleteCustomer(CustomerRequest) returns (SuccessResponse);
}
message CustomerRequest {
  int64 customerId = 1;
}

message CustomerResponse {
  int64 customerId = 1;
  string name = 2;
  string firstname = 3;
  string email = 4;
  string street = 5;
  string city = 6;
}

message Customer {
  int64 customerId = 1;
  string firstname = 2;
  string name = 3;
  string email = 4;
  string street = 5;
  string city = 6;
}

message CustomerListResponse {
  repeated Customer customers = 1;
}

message CreateCustomerRequest {
  string name = 1;
  string firstname = 2;
  string email = 3;
  string street = 4;
  string city = 5;
}

message UpdateCustomerRequest {
  int64 customerId = 1;
  string name = 2;
  string firstname = 3;
  string email = 4;
  string street = 5;
  string city = 6;
}
